solr/contrib/ltr/src/java/org/apache/solr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -67,13 +68,26 @@
 
   private String loggingModelName = DEFAULT_LOGGING_MODEL_NAME;
 
+  private LTRThreadModule threadManager = null;
+
   public void setLoggingModelName(String loggingModelName) {
     this.loggingModelName = loggingModelName;
   }
 
   @Override
   public void init(@SuppressWarnings("rawtypes") NamedList args) {
     super.init(args);
+
+    final NamedList threadManagerArgs = LTRThreadModule.extractThreadModuleParams(args);
+    // if and only if there are thread module params then we want a thread module!
+    if (threadManagerArgs.size() > 0) {
+      // create and initialize the new instance
+      threadManager = new LTRThreadModule();
```

Comment:
The ThreadModeul is only constructoed once on startup, not for every request.  The code would be much simpler as well


LTRScoringQuery:
`this.ltrThreadMgr = ltrThreadMgr;`
 `this.querySemaphore = this.ltrThreadMgr.createQuerySemaphore();`
  
LTRFeatureLoggerTransformerFactory &&
LTRQParserPlugin:
`  this.threadManager = new LTRThreadModule(args);`